examples/src/main/scala/fluence/codec/examples/KryoCodecExample.scala

```diff
@@ -0,0 +1,65 @@
+package fluence.codec.examples
+
+import cats.Id
+import fluence.codec.kryo.KryoCodecs
+import fluence.codec.{CodecError, PureCodec}
+import monix.eval.Task
+import shapeless.{::, HNil}
+
+object KryoCodecExample {
+  case class Aircraft(manufacturer: String, model: String, tailNumber: String)
+  case class Fuel(amount: Double) extends AnyVal
+
+  case class UnknownClass(x: String)
+
+  def main(args: Array[String]): Unit = {
+    // This way we can define a typed collection of codecs using kryo for the underlying serialization.
+    //
+    // These codecs can be used only to transform the corresponding type: i.e. it won't be possible to
+    // use an aircraft codec to serialize fuel (which is essentially a typed wrapper over double value).
+    //
+    // It won't be possible to obtain from this collection a codec for previously not registered class.
+    // Type safety FTW!
+    //
+    // Note that different methods are used to register Aircraft and Fuel – that's because one is a reference,
+    // and another is a value type.
+    val codecs: KryoCodecs[Task, ::[Fuel, ::[Aircraft, ::[Array[Byte], ::[Long, ::[String, HNil]]]]]] = KryoCodecs()
+      .addCase(classOf[Aircraft])
+      .add[Fuel]
+      .build[Task]()
```

Comment:
I am not sure how this `Task` affects things later on. 
Doesn't look like it's going to be pushed as a monad into the codecs being built...